src/go/rpk/pkg/cli/cluster/partitions/move.go

```diff
@@ -53,115 +54,37 @@ func newMovePartitionReplicasCommand(fs afero.Fs, p *config.Params) *cobra.Comma
-							} else {
-								ns = nt[0]
-								topic = nt[1]
-							}
-
-							current, err := cl.GetPartition(egCtx, ns, topic, part)
-							out.MaybeDie(err, "unable to get partition: %s\n", err)
-
-							newReplicas, err := configureReplicas(partition, current.Replicas)
-							out.MaybeDie(err, "unable to configure new replicas: %v\n", err)
-
-							mu.Lock()
-							newAssignmentList = append(newAssignmentList, newAssignment{
-								Namespace:   ns,
-								Topic:       topic,
-								Partition:   part,
-								OldReplicas: current.Replicas,
-								NewReplicas: newReplicas,
-							})
-							mu.Unlock()
-						}
-					} else { // -p foo/0:1,2,3
-						ns, topic, part, err := extractNTP("", partition)
-						out.MaybeDie(err, "failed to extract topic/partition: %s\n", err)
-
-						current, err := cl.GetPartition(egCtx, ns, topic, part)
-						out.MaybeDie(err, "unable to get partition: %s\n", err)
-
-						newReplicas, err := configureReplicas(partition, current.Replicas)
-						out.MaybeDie(err, "unable to configure new replicas: %v\n", err)
-
-						mu.Lock()
-						newAssignmentList = append(newAssignmentList, newAssignment{
-							Namespace:   ns,
-							Topic:       topic,
-							Partition:   part,
-							OldReplicas: current.Replicas,
-							NewReplicas: newReplicas,
-						})
-						mu.Unlock()
-					}
-					return nil
-				})
-			}
-
-			if err := g.Wait(); err != nil {
-				out.Die("failed to parse the arguments: %v\n", err)
-			}
+			newAssignmentList, coreAssignmentList, err := parseAssignments(cmd.Context(), cl, partitionsFlag, topics)
+			out.MaybeDie(err, "unable to parse new assignments: %v", err)
 
-			for _, newa := range newAssignmentList {
-				for _, nr := range newa.NewReplicas {
-					if nr.Core == -1 {
-						brokerReqs[nr.NodeID] = struct{}{}
-					}
-				}
-			}
-			for node := range brokerReqs {
-				node := node
-				g.Go(func() error {
-					broker, err := cl.Broker(cmd.Context(), node)
-					mu.Lock()
-					defer mu.Unlock()
-					knownNodeCore[node] = broker.NumCores
-					return err
-				})
-			}
+			coreAssignmentOn, err := isLocalCoreAssignmentOn(cmd.Context(), cl)
+			out.MaybeDie(err, "unable to determine if node_local_core_assignment feature is active: %v", err)
+			zap.L().Sugar().Debugf("feature node_local_core_assignment, active: %v", coreAssignmentOn)
 
-			if err := g.Wait(); err != nil {
-				out.Die("unable to find core counts", err)
+			// If core assignment is either off, or not available, we fill the
+			// assignments the old way.
+			if !coreAssignmentOn {
+				newAssignmentList, err = fillAssignmentList(cmd.Context(), cl, newAssignmentList)
+				out.MaybeDie(err, "unable to parse assignment list with existing partition replicas: %v", err)
 			}
-
-			rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+			var (
+				wg sync.WaitGroup
+				mu sync.Mutex
+			)
 			for i, newa := range newAssignmentList {
-				i := i
-				newa := newa
+				i, newa := i, newa
```

Comment:
is there a reason for the reassignment here?

Reply:
or rather, is there a reason to reassign to a local with the same name?

Reply:
An old habit of avoiding mistakingly sharing the loop variable across different routines: https://go.dev/blog/loopvar-preview (also, this original code was written before Go 1.22)

This is fixed in Go 1.22 though so we _should_ be fine removing it from the codebase (we have other instances of this across rpk), but if its ok for you, I will leave this one here until we have proper ducktape tests for this particularly dangerous command.

Reply:
That I'm familiar with, I guess I'm just used to renaming the loop variables when I do it.